src/jaxsim/rbda/kinematic_constraints.py

```diff
@@ -121,28 +167,180 @@ def compute_constraint_baumgarte_term(
     return baumgarte_term
 
 
-def compute_constraint_transforms(
+@jax.jit
+@js.common.named_scope
+def compute_constraint_wrenches(
     model: js.model.JaxSimModel,
     data: js.data.JaxSimModelData,
-    constraint: ConstraintMap,
+    *,
+    joint_force_references: jtp.VectorLike | None = None,
+    link_forces_inertial: jtp.MatrixLike | None = None,
+    regularization: jtp.Float = 1e-3,
 ) -> jtp.Matrix:
     """
-    Compute the transformation matrices for a given kinematic constraint between two frames.
+    Compute the constraint wrenches for kinematic constraints.
+
+    This function solves the constraint forces needed to satisfy kinematic constraints
+    between pairs of frames. It uses the Baumgarte stabilization method and computes
+    the constraint wrenches in inertial representation.
 
     Args:
         model: The JaxSim model.
-        data: The data of the considered model.
-        constraint: The considered constraint.
+        data: The model data.
+        joint_force_references: Optional joint force/torque references to apply. If None,
+                               zero forces are used.
+        link_forces_inertial: Optional link forces applied in inertial representation.
+                             If None, zero forces are used.
+        regularization: Regularization parameter for the constraint solver to improve
+                       numerical stability. Default is 1e-3.
 
     Returns:
-        A matrix containing the tuple of transformation matrices of the two frames.
+        Array with shape (n_constraints, 2, 6) containing constraint wrench pairs
+        in inertial representation. Each constraint produces two equal and opposite
+        wrenches applied to the constrained frames.
     """
 
-    W_H_F1 = js.frame.transform(
-        model=model, data=data, frame_index=constraint.frame_idxs_1
+    # Retrieve the kinematic constraints, if any.
+    kin_constraints = model.kin_dyn_parameters.constraints
+
+    n_kin_constraints = (
+        0
+        if (kin_constraints is None) or (kin_constraints.frame_idxs_1.shape[0] == 0)
+        else 6 * kin_constraints.frame_idxs_1.shape[0]
+    )
```

Comment:
Putting the positive case first usually helps with readability. We also use `.size` instead of `.shape[0]` at that point

```suggestion
    n_kin_constraints = (
        6 * kin_constraints.frame_idxs_1.shape[0]
        if kin_constraints is not None and kin_constraints.frame_idxs_1.size > 0
        else 0
    )
```

Reply:
Done in https://github.com/ami-iit/jaxsim/pull/452/commits/0f73dca3672581f0894711d006da7644f5c976d1